src/handlers.jl

```diff
@@ -49,6 +49,76 @@ Base.chr2ind(m::Msg, str::String, ic::Integer) = ic == 0 ? 0 :
     VersionNumber(m.header["version"]) ≥ v"5.2" ? chr2ind(str, ic) : utf16_to_ind(str, ic)
 Base.ind2chr(m::Msg, str::String, i::Integer) = i == 0 ? 0 :
     VersionNumber(m.header["version"]) ≥ v"5.2" ? ind2chr(str, i) : ind_to_utf16(str, i)
+#Compact display of types for Jupyterlab completion
+
+if isdefined(Main, :REPLCompletions)
+    import REPLCompletions: sorted_keywords, emoji_symbols, latex_symbols
+else
+    const sorted_keywords = [
+        "abstract type", "baremodule", "begin", "break", "catch", "ccall",
+        "const", "continue", "do", "else", "elseif", "end", "export", "false",
+        "finally", "for", "function", "global", "if", "import",
+        "let", "local", "macro", "module", "mutable struct",
+        "primitive type", "quote", "return", "struct",
+        "true", "try", "using", "while"]
+    import Base.REPL.REPLCompletions: emoji_symbols, latex_symbols
+end
+
+complete_type(::Type{<:Function}) = "function"
+complete_type(::Type{<:Type}) = "type"
+complete_type(::Type{<:Tuple}) = "tuple"
+
+function complete_type(T::DataType)
+    s = string(T)
+    (Compat.textwidth(s) ≤ 20 || isempty(T.parameters)) && return s
+    buf = IOBuffer()
+    print(buf, T.name)
+    position(buf) > 19 && return String(take!(buf))
+    print(buf, '{')
+    comma = false
+    for p in T.parameters
+        s = string(p)
+        if position(buf) + sizeof(s) > 20
+            comma || print(buf, '…')
+            break
+        end
+        comma && print(buf, ',')
+        comma = true
+        print(buf, s)
+    end
+    print(buf, '}')
+    return String(take!(buf))
+end
+
+#Get typeMap for Jupyter completions
+function complete_types(comps)
+    typeMap = []
+    for c in comps
+        ctype = ""
+        if !isempty(searchsorted(sorted_keywords, c))
+            ctype = "keyword"
+        elseif startswith(c, "\\:")
+            ctype = get(emoji_symbols, c, "")
+            isempty(ctype) || (ctype = "emoji: $ctype")
+        elseif startswith(c, "\\")
+            ctype = get(latex_symbols, c, "")
+        else
+            expr = Meta.parse(c, raise=false)
+            if typeof(expr) == Symbol
+                try
+                    ctype = complete_type(eval(Main, :(typeof($expr))))
+                catch
```

Comment:
You can just omit `catch` if you have an empty catch block.

Reply:
OK, I removed it now.